code/code/cmd/cmd_stab.cc

```diff
@@ -1,345 +1,239 @@
-      // Waist :  2 (can not be whacked)
-      // Legs   :  1 (can not be whacked)
-      // Fingers: -4 (can     be whacked)
-      if (::number(tNewSever, 500) <= 0 &&
-          !tSucker->isLucky(tSucker->GetMaxLevel())) {
-        tStringChar = format("You slice $N's %s right off!") % tStLimb;
-        tStringVict = format("$n slices your %s right off!") % tStLimb;
-        tStringOthr="$n slices $N's %s right off!") % tStLimb;
-
-        tSucker->makePartMissing(tLimb, false, tThief);
-
-        act(tStringChar, FALSE, tThief, NULL, tSucker, TO_CHAR);
-        act(tStringVict, FALSE, tThief, NULL, tSucker, TO_VICT);
-        act(tStringOthr, FALSE, tThief, NULL, tSucker, TO_NOTVICT);
+      // New bleed
+      sstring limbName(victim->describeBodySlot(limb));
+      act(format("Your stab opens a <r>bleeding wound<z> on $N's %s!") % limbName,
+        FALSE, thief, nullptr, victim, TO_CHAR);
+      act(format("$n's stab opens a <r>bleeding wound<z> on your %s!") % limbName,
+        FALSE, thief, nullptr, victim, TO_VICT);
+      act(format("$n's stab opens a <r>bleeding wound<z> on $N's %s!") % limbName,
+        FALSE, thief, nullptr, victim, TO_NOTVICT);
+
+      victim->rawBleed(limb, duration, SILENT_YES, CHECK_IMMUNITY_NO);
+    }
+  }
+}
+
+// Offhand stab attack - uses SKILL_STABBING for specAttack
+static int stabOffAtk(TBeing* thief, TBeing* victim, TGenWeapon* weapon) {
+  if (!weapon || !weapon->canStab())
+    return FALSE;
+
+  // Use DEX/SPE for offense, AGI/PER for defense
+  int specResult = thief->specialAttack(victim, SKILL_STABBING, 0, STAT_DEX,
+    STAT_SPE, STAT_AGI, STAT_PER, false);
+  if (specResult != COMPLETE_SUCCESS && specResult != GUARANTEED_SUCCESS) {
+    stabMissMsg(thief, victim, weapon);
+    victim->addHated(thief);
+    return TRUE;
+  }
+
+  // Calculate damage for offhand
+  int level = thief->getSkillLevel(SKILL_STABBING);
+  int dam = thief->getSkillDam(victim, SKILL_STABBING, level,
+    thief->getAdvLearning(SKILL_STABBING));
+  dam = thief->getActualDamage(victim, weapon, dam, SKILL_STABBING);
+  
+  if (dam <= 0) {
+    act("Your $o fails to penetrate $N's thick hide.", FALSE, thief, weapon,
+      victim, TO_CHAR);
+    act("$n's $o fails to penetrate your thick hide.", FALSE, thief, weapon,
+      victim, TO_VICT);
+    act("$n's $o fails to penetrate $N's thick hide.", FALSE, thief, weapon,
+      victim, TO_NOTVICT);
+    return TRUE;
+  }
+
+  // Get target limb
+  wearSlotT limb = victim->getPartHit(thief, FALSE);
+  if (!victim->hasPart(limb))
+    limb = victim->getCritPartHit();
+
+  // Show hit message
+  stabHitMsg(thief, victim, weapon, limb);
+
+  // Apply damage
+  if (thief->reconcileDamage(victim, dam, SKILL_STABBING) == -1)
+    return DELETE_VICT;
+
+  // Check for bleeding and poison
+  stabBleedCheck(thief, victim, weapon, limb);
+  stabPoisonCheck(victim, weapon);
+
+  // Check weapon spec proc
+  if (weapon->checkSpec(victim, CMD_STAB, reinterpret_cast<char*>(limb),
```

Comment:
nit, beyond the scope of your PR:

> `reinterpret_cast<char*>(limb)`

The traditional way of doing this was to pass `void*`, the modern is `std::variant`. `char*` is bad because it affords printing/logging/similar.